Code/GraphMol/FileParsers/GeneralFileReader.h

```diff
@@ -0,0 +1,233 @@
+//
+//  Copyright (C) 2020 Shrey Aryan
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#ifndef GENERAL_FILE_READER_H
+#define GENERAL_FILE_READER_H
+#include <string>
+#include <iostream>
+#include <vector>
+#include "MolSupplier.h"
+#include <RDGeneral/BadFileException.h>
+#include <RDStreams/streams.h>
+
+namespace RDKit {
+struct SupplierOptions {
+  bool takeOwnership = true;
+  bool sanitize = true;
+  bool removeHs = true;
+  bool strictParsing = true;
+
+  std::string delimiter = "\t";
+  int smilesColumn = 0;
+  int nameColumn = 1;
+  bool titleLine = true;
+
+  std::string nameRecord = "";
+  int confId2D = -1;
+  int confId3D = 0;
+};
+
+//! determines whether file and compression format are valid
+bool valid(std::string& fileFormat, std::string& compressionFormat) {
+  //! current formats supported
+  std::vector<std::string> fileFormats{"sdf", "mae", "smi", "csv",
```

Comment:
I would suggest moving the definitions of these outside the function, so that they become global variables in the `RDKit::GeneralMolSupplier` namespace, and renaming them to `supportedFileFormats` and `supportedCompressionFormats`

Reply:
when you do that, please be sure to add `maegz` to the list of supported file formats.